shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/SchemaBuilder.java

```diff
@@ -38,45 +37,38 @@
 public final class SchemaBuilder {
     
-    static {
-        ShardingSphereServiceLoader.register(DialectTableMetaDataLoader.class);
-    }
-    
     /**
-     * build actual and logic table meta data.
+     * Build sharding sphere schema.
      *
      * @param materials schema builder materials
-     * @return actual and logic table meta data
+     * @return sharding sphere schema
      * @throws SQLException SQL exception
      */
-    public static Map<TableMetaData, TableMetaData> build(final SchemaBuilderMaterials materials) throws SQLException {
-        Map<String, TableMetaData> actualTableMetaMap = buildActualTableMetaDataMap(materials);
-        Map<String, TableMetaData> logicTableMetaMap = buildLogicTableMetaDataMap(materials, actualTableMetaMap);
-        Map<TableMetaData, TableMetaData> result = new HashMap<>(actualTableMetaMap.size(), 1);
-        for (Entry<String, TableMetaData> entry : actualTableMetaMap.entrySet()) {
-            result.put(entry.getValue(), logicTableMetaMap.getOrDefault(entry.getKey(), entry.getValue()));
-        }
-        return result;
-    }
-    
-    private static Map<String, TableMetaData> buildActualTableMetaDataMap(final SchemaBuilderMaterials materials) throws SQLException {
+    public static ShardingSphereSchema build(final SchemaBuilderMaterials materials) throws SQLException {
         Collection<String> allTableNames = materials.getRules().stream().filter(each -> each instanceof TableContainedRule)
                 .flatMap(shardingSphereRule -> ((TableContainedRule) shardingSphereRule).getTables().stream()).collect(Collectors.toSet());
-        return TableMetaDataBuilder.load(allTableNames, materials);
+        return new ShardingSphereSchema(TableMetaDataBuilder.load(allTableNames, materials));
     }
     
-    private static Map<String, TableMetaData> buildLogicTableMetaDataMap(final SchemaBuilderMaterials materials, final Map<String, TableMetaData> tables) {
-        Map<String, TableMetaData> result = new HashMap<>(materials.getRules().size(), 1);
+    /**
+     * Decorate sharding sphere schema.
+     *
+     * @param schema sharding sphere schema
+     * @param materials schema builder materials
+     * @return sharding sphere schema
+     */
+    public static ShardingSphereSchema decorateSchema(final ShardingSphereSchema schema, final SchemaBuilderMaterials materials) {
+        Map<String, TableMetaData> tableMetaDataMap = new LinkedHashMap<>(schema.getTables());
         for (ShardingSphereRule rule : materials.getRules()) {
             if (rule instanceof TableContainedRule) {
                 for (String table : ((TableContainedRule) rule).getTables()) {
-                    if (tables.containsKey(table)) {
-                        TableMetaData metaData = TableMetaDataBuilder.decorate(table, tables.get(table), materials.getRules());
-                        result.put(table, metaData);
+                    if (tableMetaDataMap.containsKey(table.toLowerCase())) {
```

Comment:
@tuichenchuxin Can we remove toLowerCase logic? We can get the actual `TableMetaData` from schema.